network/network.go

```diff
@@ -227,6 +241,34 @@ func (c *Collector) Collect(ch chan<- prometheus.Metric) {
 	c.logger.Log("level", "info", "message", "collected metrics", "scrapeID", c.scrapeID, "scrapeTime", scrapingElapsed.Seconds())
 }
 
+func (c *Collector) podExists(podIP string, podList *v1.PodList) bool {
+	podName, ok := c.podNameFromIP(podIP, podList)
+	if !ok {
+		return false
+	}
+
+	// Get the Pod to see if it still exists
+	_, err := c.k8sClient.CoreV1().Pods(c.namespace).Get(podName, metav1.GetOptions{})
```

Comment:
You can get the response here and check if `deletionTimestamp` is set - if it is set then the pod is in a deleting state and might already no have an IP in the service anymore.